core/sync.js

```diff
@@ -112,73 +111,51 @@ class Sync {
   // First, start metadata synchronization in pouch, with the watchers
   // Then, when a stable state is reached, start applying changes from pouch
   async start() /*: Promise<void> */ {
-    if (this.started && (await this.started)) return
-
-    let runningResolve, runningReject
-    this.running = new Promise((resolve, reject) => {
-      runningResolve = resolve
-      runningReject = reject
-    })
-
-    this.started = new Promise(async (resolve, reject) => {
-      if (this.stopRequested) {
-        reject()
-        runningReject()
-        return
-      }
-
-      log.info('Starting Sync...')
+    if (this.lifecycle.willStop()) {
+      await this.lifecycle.stopped()
+    } else {
+      return
+    }
 
-      try {
-        await this.local.start()
-        const localRunning = this.local.watcher.running
-        const {
-          running: remoteRunning,
-          started: remoteStarted
-        } = this.remote.start()
-        await remoteStarted
-
-        Promise.all([localRunning, remoteRunning]).catch(err => {
-          throw err
-        })
-      } catch (err) {
-        log.error({ err }, 'Could not start watchers')
-        this.local.stop()
-        this.remote.stop()
-        reject(err)
-        runningReject(err)
-      }
-      resolve(true)
+    try {
+      this.lifecycle.begin('start')
+    } catch (err) {
+      return
+    }
+    try {
+      await this.local.start()
+      await this.remote.start()
+    } catch (err) {
+      this.error(err)
+      this.lifecycle.end('start')
+      await this.stop()
+      return
+    }
+    this.lifecycle.end('start')
+
+    Promise.all([
+      this.local.watcher.running,
+      this.remote.watcher.running
+    ]).catch(err => {
+      this.error(err)
+      this.stop()
```

Comment:
No `await` here?

Reply:
We're not awaiting the two promises here so we're not await the `catch` handler either so awaiting `this.stop()` would not change anything.